example.ipynb

```diff
@@ -213,1997 +404,504 @@
-      "                  'derived_from_specimen': 'f0c10dc4-a6d6-4372-a3a5-af61648cd09b',\n",
-      "                  'derived_from_subject': 'TCGA-3P-A9WA',\n",
-      "                  'id': '3280e41e-de2e-4a4a-b7b7-06fd06def017',\n",
-      "                  'identifier': [ { 'system': 'GDC',\n",
-      "                                    'value': '3280e41e-de2e-4a4a-b7b7-06fd06def017'}],\n",
-      "                  'primary_disease_type': 'Cystic, Mucinous and Serous '\n",
-      "                                          'Neoplasms',\n",
-      "                  'source_material_type': 'Primary Tumor',\n",
-      "                  'specimen_type': 'aliquot'},\n",
-      "                { 'File': [],\n",
-      "                  'age_at_collection': None,\n",
-      "                  'anatomical_site': None,\n",
-      "                  'associated_project': 'TCGA-OV',\n",
-      "                  'derived_from_specimen': 'f0c10dc4-a6d6-4372-a3a5-af61648cd09b',\n",
-      "                  'derived_from_subject': 'TCGA-3P-A9WA',\n",
-      "                  'id': '94e91d8c-244a-48cf-af09-9412659e6759',\n",
-      "                  'identifier': [ { 'system': 'GDC',\n",
-      "                                    'value': '94e91d8c-244a-48cf-af09-9412659e6759'}],\n",
-      "                  'primary_disease_type': 'Cystic, Mucinous and Serous '\n",
-      "                                          'Neoplasms',\n",
-      "                  'source_material_type': 'Primary Tumor',\n",
-      "                  'specimen_type': 'aliquot'}],\n",
-      "  'Treatment': [ {'outcome': None, 'type': 'Radiation Therapy, NOS'},\n",
-      "                 {'outcome': None, 'type': 'Pharmaceutical Therapy, NOS'}],\n",
-      "  'age_at_diagnosis': 20211,\n",
-      "  'associated_project': 'TCGA-OV',\n",
-      "  'days_to_birth': -20211,\n",
-      "  'ethnicity': 'not hispanic or latino',\n",
-      "  'id': 'a9e4580c-412c-5325-a91d-50fd125a515b',\n",
-      "  'identifier': [ { 'system': 'GDC',\n",
-      "                    'value': 'e35b2813-427e-4e83-95f6-4f0281f42a59'}],\n",
-      "  'morphology': '8440/3',\n",
-      "  'primary_diagnosis': 'Cystadenocarcinoma, NOS',\n",
-      "  'primary_disease_site': 'Ovary',\n",
-      "  'primary_disease_type': 'Cystic, Mucinous and Serous Neoplasms',\n",
-      "  'race': 'black or african american',\n",
-      "  'sex': 'female',\n",
-      "  'tumor_grade': 'not reported',\n",
-      "  'tumor_stage': 'not reported'}\n"
+      "{'TCGA-UVM', 'TCGA-SKCM', 'FM-AD'}\n"
      ]
     }
    ],
    "source": [
-    "r.pretty_print(0)"
+    "project_list = set()\n",
+    "\n",
+    "for patient in r:\n",
+    "    research_subjects = patient['ResearchSubject']\n",
+    "    for rs in research_subjects:\n",
+    "        project = rs['associated_project']\n",
+    "        if project not in project_list:\n",
+    "            project_list.add(project)\n",
```

Comment:
Since `project_list` is a set you don't need to check for membership before adding
```suggestion
    "        project_list.add(rs['associated_project'])\n",
```

Reply:
Thank you for doing this 🙏